### ISSUE TITLE
http://debian.notset.fr/snapshot/archive/debian/ works and http://debian.notset.fr/snapshot/archive/debian doesn't

### ISSUE THREAD

**Issue by safinaskar** [fepitre/debian-snapshot#14]
URL http://debian.notset.fr/snapshot/archive/debian/ works and URL http://debian.notset.fr/snapshot/archive/debian doesn't. Here is wget output:

```text
d-user@comp:/tmp$ wget http://debian.notset.fr/snapshot/archive/debian/
--2022-10-22 18:12:17--  http://debian.notset.fr/snapshot/archive/debian/
Resolving debian.notset.fr (debian.notset.fr)... 80.11.163.215
Connecting to debian.notset.fr (debian.notset.fr)|80.11.163.215|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘index.html’

index.html                                  [      <=>                                                                       ] 818,82K   536KB/s    in 1,5s    

2022-10-22 18:12:19 (536 KB/s) - ‘index.html’ saved [838475]

d-user@comp:/tmp$ wget http://debian.notset.fr/snapshot/archive/debian
--2022-10-22 18:12:21--  http://debian.notset.fr/snapshot/archive/debian
Resolving debian.notset.fr (debian.notset.fr)... 80.11.163.215
Connecting to debian.notset.fr (debian.notset.fr)|80.11.163.215|:80... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: http://debian.notset.fr/archive/debian/ [following]
--2022-10-22 18:12:21--  http://debian.notset.fr/archive/debian/
Reusing existing connection to debian.notset.fr:80.
HTTP request sent, awaiting response... 404 Not Found
2022-10-22 18:12:21 ERROR 404: Not Found.
```

P. S. Thanks a lot for service. I plan to use it. It would be very cool if you also add `stretch` suite and also `stretch-updates` and `stretch/updates`

**Comment by rclobus** [comment]
Hello safinaskar,

I have 2 questions:
1) Which tool/configuration are you using that you need to access `archive/debian` without the trailing slash?

2) Why would you want to use snapshots for `stretch`? That is `oldoldstable`, you'll find the repository on archive.debian.org and (since there will be no updates) you don't need to use the snapshot server for that.

With kind regards,
Roland Clobus

**Comment by safinaskar** [comment]
1. Trailing slash doesn't matter for my tools. This is question of taste. I think that every time web server process `a` and `a/` differently is a bug.
2. My main OS is debian stretch. I'm writing tool for creating reproducible images for reproducing bugs. Currently I have to have this in my code:
```bash
  if [ "$CODENAME" = stretch ]; then
    MIRROR="http://deb.debian.org/debian"
  else
    MIRROR="http://snapshot.notset.fr/archive/debian/$TIMESTAMP"
  fi
```
If notset.fr have stretch, I will be able to delete this conditional. But, of course, feel free to keep stretch out of archive, it will be OK for me to keep this conditional

**Comment by rclobus** [comment]
Please note that Debian Stretch was end-of-life on 2020-07-18, more than 2 years ago.

I've done a lot regarding reproducible live images, see my Wiki page: https://wiki.debian.org/ReproducibleInstalls/LiveImages
In the git-repository of live-build you'll find in the test folder 'rebuild.sh', which probably already does a lot of the things you might want to do.

Bookworm has reproducible live images, without any special hooks, and bullseye needs only a few of them.
Automated tests are already running on Debian architecture: https://jenkins.debian.net/view/live/ and https://openqa.debian.net/group_overview/14

**Comment by safinaskar** [comment]
I used slightly incorrect term. I don't need reproducible images (i. e. bit-to-bit identical images). I need images, which are similar to "real" ones (i. e. installed using normal d-i on default settings) as closely as possible. Currently I use my thin wrapper around debootstrap, then I plan to switch to thin wrapper around automated d-i. I don't trust not-so-widely-used tools such as live-build, debuerreotype, mmdebstrap, because they introduce their own modifications to images (as opposed to d-i or debootstrap).

I need all this for reproducing bugs.

Using live images sometimes introduces their own bugs, such as this: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=832999 (I found this bug a lot of time ago, I hope it is fixed now)
